var/spack/repos/builtin/packages/perl-fth/package.py

```diff
@@ -0,0 +1,122 @@
+
+    variant('hevea', default=False,
+            description="Use hevea when inputting LaTeX files (fth.pl -hevea)")
+    variant('pdflatex', default=False,
+            description="Use pdflatex to make a LaTeX index file (fth.pl -latexindex)")
+
+    depends_on('perl', type='run')
+    depends_on('perl-cgi', type='run')
+    # Actual dependency was on etags only, but no longer in recent releases:
+    depends_on('emacs', type='run', when='@:0.520')
+    # For fth.pl -hevea option
+    depends_on('hevea', when='+hevea', type='run')
+    # Actual dependency is on pdflatex only for fth.pl -latexindex option
+    depends_on('texlive', when='+pdflatex', type='run')
+    # initmak.pl uses md5sum provided by coreutils
+    depends_on('coreutils', type='run')
+    depends_on('dos2unix', type='build')
+
+    # Patches to remove the ancient ksh shebang for fth.pl and initmak.pl.
+    # git diff a/bin/fth.pl b/bin/fth.pl
+    patch('fth-shebang.patch', when='@0.517:0.522', sha256='3e82d34c8ae1709e5480fac87db387c1c2e219d7b7d596c8a9d62f0da2439ab3')
+    patch('fth-shebang2.patch', when='@0.517:0.522', sha256='839be7c0efad752ae341379c81ee1df4a3a81f608f802998c6b4ebc4bae8e167')
+
+    def _make_executable(self, name):
+        return Executable(join_path(self.prefix.bin, name + '.pl'))
+
+    def setup_dependent_package(self, module, dependent_spec):
+        # https://spack-tutorial.readthedocs.io/en/latest/tutorial_advanced_packaging.html
+        checks = ['fth', 'getin', 'view', 'javatex2', 'tex3ht', 'initmak']
+        for name in checks:
+            setattr(module, name, self._make_executable(name))
+
+    def setup_dependent_build_environment(self, env, dependent_spec):
+        env.set('JAVATEX_DIR', self.prefix)
+        env.set('FTAGSHTML_DIR', self.prefix)
+        env.set('FTAGSHTML_DOC', join_path(self.prefix, 'doc'))
+
+    def setup_run_environment(self, env):
+        # https://github.com/spack/spack/discussions/13926
+        # Let us set the adequate environment when loading perl-fth
+        env.set('JAVATEX_DIR', self.prefix)
+        env.set('FTAGSHTML_DIR', self.prefix)
+        env.set('FTAGSHTML_DOC', join_path(self.prefix, 'doc'))
+
+    def install(self, spec, prefix):
+        # Remove the perl shebang with the local perl
+        # (since ftagshtml has no Makefile.PL to do it).
+        checks = ['fth', 'getin', 'view', 'javatex2', 'tex3ht', 'initmak']
+        # mstr = '#!' + join_path(spec['perl'].prefix.bin, 'perl')
+        mstr = '#!' + spec['perl'].command.path
+        with working_dir('bin'):
+            for exe in checks:
+                fic = exe + '.pl'
+                if os.path.exists(fic):
+                    dos2unix = which('dos2unix')
+                    dos2unix(fic)
+                    fthfile = FileFilter(fic)
+                    fthfile.filter('#!/usr/bin/perl', mstr)
+                    fthfile.filter('#!/usr/bin/env perl', mstr)
```

Comment:
You might want to turn off backups to avoid having backup files in the install directory.
```suggestion
                    fthfile.filter('#!/usr/bin/perl', mstr, backup=False)
                    fthfile.filter('#!/usr/bin/env perl', mstr, backup=False)
```